bot/cogs/translate_cog.py

```diff
@@ -0,0 +1,196 @@
+import json
+import logging
+import discord
+import aiohttp
+import uuid
+import bot.extensions as ext
+import discord.ext.commands as commands
+from discord.ext.commands.errors import UserInputError
+from bot.messaging.events import Events
+from bot.bot_secrets import BotSecrets
+from bot.consts import Colors
+
+log = logging.getLogger(__name__)
+
+LANGUAGE_NAME_TO_SHORT_CODE = {
+    "afrikaans": "af",
+    "arabic": "ar",
+    "bulgarian": "bg",
+    "catalan": "ca",
+    "chinese simplified": "zh-hans",
+    "chinese traditional":	"zh-hant",
+    "croatian": "hr",
+    "czech": "cs",
+    "danish": "da",
+    "dutch": "nl",
+    "english": "en",
+    "estonian": "et",
+    "finnish": "fi",
+    "french": "fr",
+    "german": "de",
+    "greek": "el",
+    "gujarati": "gu",
+    "haitian creole": "ht",
+    "hebrew": "he",
+    "hindi": "hi",
+    "hungarian":"hu",
+    "icelandic": "is",
+    "indonesian": "id",
+    "irish": "ga",
+    "italian": "it",
+    "japanese": "ja",
+    "klingon": "tlh-Latn",
+    "korean": "ko",
+    "kurdish (central)": "ku-arab",
+    "latvian": "lv",
+    "lithuanian": "lt",
+    "malay": "ms",
+    "maltese": "mt",
+    "norwegian": "nb",
+    "pashto": "ps",
+    "persian": "fa",
+    "polish": "pl",
+    "portuguese": "pt",
+    "romanian": "ro",
+    "russian": "ru",
+    "serbian (cyrillic)": "sr-cyrl",
+    "serbian (latin)": "sr-catn",
+    "slovak": "sk",
+    "slovenian": "sl",
+    "spanish": "es",
+    "swahili": "sw",
+    "swedish": "sv",
+    "tahitian": "ty",
+    "thai": "th",
+    "turkish": "tr",
+    "ukrainian": "uk",
+    "urdu": "ur",
+    "vietnamese": "vi",
+    "welsh": "cy",
+    "yucatec maya": "yua"
+}
+CHUNK_SIZE = 15
+LANGUAGE_SHORT_CODE_TO_NAME = {value : key for key, value in LANGUAGE_NAME_TO_SHORT_CODE.items()}
+
+HEADERS = {
+    'Ocp-Apim-Subscription-Key': BotSecrets.get_instance().azure_translate_key,
+    'Ocp-Apim-Subscription-Region': 'global',
+    'Content-type': 'application/json',
+    'X-ClientTraceId': str(uuid.uuid4())
+}
+
+
+TRANSLATE_API_URL = "https://api.cognitive.microsofttranslator.com/translate"
+
+class TranslateCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @ext.command()
+    @ext.long_help('Allows you to translate words or sentences by either specifying both the input and output language with the text to translate, or just the output language and the text to translate')
+    @ext.short_help('Translates words or phrases between two languages')
+    @ext.example(('translate en spanish Hello', 'translate german Hello'))
```

Comment:
we should probably add some more examples tbqh, its a complex feature so people will need help

Reply:
aight, will do